lib/kotkowo_web/live/news_live/found_home.ex

```diff
@@ -0,0 +1,389 @@
+      |> assign(:max_page, max_page)
+
+    {:noreply, socket}
+  end
+
+  @impl true
+  def handle_event(
+        "filter_cats",
+        %{
+          "_target" => event,
+          "date_from" => date_from,
+          "date_to" => date_to,
+          "cat_search" => name_query,
+          "tag_search" => tag_query
+        } = params,
+        socket
+      ) do
+    limit = socket.assigns |> Map.get(:limit, Integer.to_string(@default_limit)) |> abs()
+    page = Map.get(socket.assigns, :page, Integer.to_string(@first_page))
+
+    not_castrated = params |> Map.get("not_castrated", nil) |> query_to_bool()
+
+    is_castrated =
+      params
+      |> Map.get("is_castrated", nil)
+      |> query_to_bool()
+      |> castrated_switch(not_castrated, event)
+
+    ages =
+      for key <- Seniority.all(),
+          true == params |> Map.get(Atom.to_string(key), false) |> query_to_bool(),
+          do: key
+
+    genders =
+      for key <- Sex.all(),
+          true == params |> Map.get(Atom.to_string(key), false) |> query_to_bool(),
+          do: key
+
+    colors =
+      for key <- Color.all(),
+          true == params |> Map.get(Atom.to_string(key), false) |> query_to_bool(),
+          do: key
+
+    query_params = %{
+      name: name_query,
+      tag: tag_query,
+      castrated: is_castrated,
+      ages: ages,
+      genders: genders,
+      date_from: date_from,
+      colors: colors,
+      date_to: date_to,
+      page: page,
+      limit: limit
+    }
+
+    socket =
+      push_patch(socket, to: ~p"/aktualnosci/znalazly-dom?#{query_params}")
+
+    {:noreply, socket}
+  end
+
+  @impl true
+  def handle_event("items_amount", %{"items_per_page" => amount}, socket) do
+    limit = amount |> String.to_integer() |> abs()
+    max_page = max(@first_page, socket.assigns.filtered_cats |> length() |> div(limit))
+
+    query_params = %{
+      name: socket.assigns.name,
+      tag: socket.assigns.tag,
+      castrated: socket.assigns.castrated,
+      ages: socket.assigns.ages,
+      colors: socket.assigns.colors,
+      genders: socket.assigns.genders,
+      date_from: socket.assigns.date_from,
+      date_to: socket.assigns.date_to,
+      page: socket.assigns.page,
+      limit: limit
+    }
+
+    socket =
+      socket
+      |> assign(:max_page, max_page)
+      |> push_patch(to: ~p"/aktualnosci/znalazly-dom?#{query_params}")
+
+    {:noreply, socket}
+  end
+
+  @impl true
+  def handle_event("select_page", %{"value" => page}, socket) do
+    query_params = %{
+      name: socket.assigns.name,
+      tag: socket.assigns.tag,
+      castrated: socket.assigns.castrated,
+      ages: socket.assigns.ages,
+      genders: socket.assigns.genders,
+      date_from: socket.assigns.date_from,
+      colors: socket.assigns.colors,
+      date_to: socket.assigns.date_to,
+      page: page,
+      limit: socket.assigns.limit
+    }
+
+    socket = push_patch(socket, to: ~p"/aktualnosci/znalazly-dom?#{query_params}")
```

Comment:
every time you `push_patch` you trigger `handle_params/3`
just move all logic of parsing and assigning to `handle_params/3`(is it not already there?) and then:

```elixir
query_params = %{socket.assigns.query_params | page: page}
socket = push_patch(to: ~p"/aktualnosci/znalazly-dom?#{query_params}")
```